### ISSUE TITLE
Format and syntax highlight hover responses from Hack LSP

### ISSUE THREAD

**Issue by PranayAgarwal** [facebook/hhvm#8247]
### HHVM Version

All

### Operating System and Version

All

### Standalone code, or other way to reproduce the problem

In VS Code, write hack code and hover over a variable.

### Actual result

Shows the variable type as plain text, and if the type is slightly complicated (e.g. a nested shape) result is hard to parse.

![image](https://user-images.githubusercontent.com/341507/42114348-95500160-7ba3-11e8-9843-4e4986f4097b.png)

### Expected result

The response could be formatted and syntax highlighted.

TypeScript example:
![image](https://user-images.githubusercontent.com/341507/42114358-a5dd4646-7ba3-11e8-91d9-0017f74d29a7.png)

**Comment by fredemmott** [comment]
This is available with `hh_client lsp --enhanced-hover`; I'll dig into why that's not an initialization option or just based on client capabilities

**Comment by fredemmott** [comment]
![screen shot 2018-09-18 at 3 44 00 pm](https://user-images.githubusercontent.com/360927/45720619-bcdec680-bb59-11e8-8588-80290243596d.png)
![screen shot 2018-09-18 at 3 36 40 pm](https://user-images.githubusercontent.com/360927/45720620-bcdec680-bb59-11e8-959e-bc412106ba03.png)


FB: D9928147
